### PR TITLE
Item Methods and Routes

### PR DESCRIPTION
**Pull request by jen-chieh**
Merge into Main request

### CODE REVIEW

File: App.ts [hunk 1]
```diff
@@ -49,15 +49,47 @@ class App {
 
     /* Restaurant Routes */
     router.get('/api/restaurants/', (req, res) => {
-      console.log('Query All Restaurants');
-      this.Restaurants.retrieveAllRestaurants(res);
+        console.log('Query All Restaurants');
+        this.Restaurants.retrieveAllRestaurants(res);
```
Comment (goyaladit) [inline App.ts:53]: very minor issue: indent is more than standard indent for this file

File: App.ts [hunk 2]
```diff
@@ -49,15 +49,47 @@ class App {
...
+    router.post('/api/createItem', (req, res) => {
+      console.log('Insert item into items collection');
+      let createItem: object = {
+        itemName:req.body.itemName,
+        itemDecription:req.body.itemDecription,
+        itemPrice:req.body.itemPrice,
+        itemImg : req.body.itemImg,
+        ItemID: crypto.randomUUID(),
+        restaurantID: req.body.restaurantId,
+        menusID:req.body.menusid
```
Comment (goyaladit) [inline App.ts:87]: should 'menusID' maybe be changed to 'menuIDs' ?

File: App.ts [hunk 3]
```diff
@@ -145,14 +175,37 @@ class App {
...
+      try{
+        // Pre check: check if the item exists in the database
+        // this.Items.retrieveItemByID(res, { itemId: itemId }); // TODO : implement this
+        // console.log("Adding " + itemName + " : " + menuId + " for restaurant: " + restaurantId);
+        // Query the database to add a section to the menu
+        const item = this.Items.getItem(res, { itemID: itemId });
+        if (item == undefined || item == null) {
+        res.status(400).json({ message: 'Item is not found' });
+        return;
+        }
```
Comment (goyaladit) [inline App.ts:187]: is there any corresponding 'catch' code block for this 'try' block?

File: App.ts [hunk 4]
```diff
@@ -145,14 +175,37 @@ class App {
...
+
+        // add menu section
+        const menuSection = this.Menus.retrieveMenuSections(res, { restaurantID: restaurantId, menuID: menuId, sectionName });
+        if (menuSection === undefined || menuSection === null) {
+        res.status(400).json({ message: 'Menu section is not found' });
+        return;
```
Comment (goyaladit) [inline App.ts:193]: minor issue: code should be indented

File: model/ItemModel.ts [hunk 5]
```diff
@@ -24,28 +24,92 @@ class ItemModel {
...
         this.model = mongooseConnection.model<IItemModel>("Items", this.schema);
     }
-
+     /**
+     * Retrieve all items in item collection
+     *@param response - response object
+     */
     public retrieveAllItems(response:any): any {
-        console.log("start query");
+        console.log("retrieve All Items");
```
Comment (goyaladit) [inline model/ItemModel.ts:38]:
```suggestion
        console.log("retrieve All Items....");
```

File: model/ItemModel.ts [hunk 6]
```diff
@@ -24,28 +24,92 @@ class ItemModel {
...
         });
     }
+    /**
+     * Retrieve a single item by ID
+     *@param response - response object
+     *@param filter - filter object
+     *@returns - JSON Object
+     */
+   public retrieveItemByID(response: any, filter: Object) {
+       // TODO: Check if this is working
```
Comment (goyaladit) [inline model/ItemModel.ts:66]: Were you able to check if this was working?

File: model/ItemModel.ts [hunk 7]
```diff
@@ -24,28 +24,92 @@ class ItemModel {
...
+       console.log("Query single item with filter: " + JSON.stringify(filter));
+       var query = this.model.findOne(filter);
+       query.exec((err, itemArray) => {
+        if (err) return console.error(err);
+          // response.json(itemArray);
+           return JSON.stringify(itemArray);
```
Comment (goyaladit) [inline model/ItemModel.ts:72]: Is this method different from the 'getItem()' method below?

**Review by goyaladit — COMMENTED** [review]
left some comments to consider

**Review by Nirmalya24 — APPROVED** [review]
LGTM

**Pull request merged**